packages/js/plugins/ethereum/src/Connections.ts

```diff
@@ -0,0 +1,125 @@
+import { Connection } from "./Connection";
+import { Connection as SchemaConnection } from "./wrap";
+
+import { getNetwork } from "@ethersproject/providers";
+
+export interface ConnectionsConfig {
+  networks: {
+    [network: string]: Connection;
+  };
+  defaultNetwork?: string;
+}
+
+// https://www.typescriptlang.org/play?pretty=true&ts=4.7.4#code/LAKAxgNghgzjAEBJAdgEwKYA92oAoCcB7ABwGUoBbYideeAb1DoG1lL0AueGAF3wEtkAc3gAfbgE8KAI0IQAulyjIJAblBN4xAQDcoPWr338w8QRmyoAEstQ18XAoUwSbaewB4UFnE7KVqdAA+eABeBk06IXQeAAoeKHxongAaLSJidHweCQBKCJA6Ivh8GIBXfGR4BKSY5m0SLJz5dUKigF8UyO4Y+MTktIbM7Ik0vQgy9FyuWTl0ZQLi4prk+oymiXkw+HHJ1qWi0p4Kqr497vbNdtbNMEJkXnwysB5CfFj8xjbio5P4ZHQAHd4E4XPEABb8GBpbxYXwZchUGgAOnMcOstnsuX28EuIE0pSgqFhlj821iQy4j0EInEMCksgg03gygkYRCXwOv0q1UhMDWJBaVw0IDxIr0+Gq6F42wBwJJ8JIiMCHxuIAMvGYACIhgBhMq8QgULVbcJa3bofWG41qu4PObIiCEITxaU8bV6g2vY3ybHwAD0-vg5qgE0tXqNWtAdpgDqdLo1PGRhOJaHRfliOoyVu9WtyfsDwYtOcj0fusZR8ddMsERmQYHQhAAZkg06SEQEaAWg2d0GX7ZXnbEAPLSABW6BeyIA1ugJDBqzx86oA0HtSmFXgMlq0lmSCWffBQEA
+export class Connections {
+  private _connections: { [network: string]: Connection } = {};
+  private _defaultNetwork: string;
+
+  constructor(config: ConnectionsConfig) {
+    for (const [network, connection] of Object.entries(config.networks)) {
+      this.set(network, connection);
+    }
+    // Assign the default network (mainnet if not provided)
+    if (config.defaultNetwork) {
+      this.setDefaultNetwork(config.defaultNetwork);
+    } else {
+      this.setDefaultNetwork("mainnet");
+    }
+  }
+
+  /** Returns Connection indexed by network name, or by default network if key is undefined */
+  get(network?: string): Connection {
+    if (!network) {
+      return this._connections[this._defaultNetwork];
+    }
+    return this._connections[network];
+  }
+
+  /** sets Connection to index of network name */
+  set(network: string, connection: Connection): void {
```

Comment:
I'm a bit concerned with how we don't have much testing that exhausts these new functions. Could we add a test file for connections + connection?

Reply:
Yes, of course. I'll update with tests asap.

Reply:
I've added tests for Connections and Connection